controllers/object_controls.go

```diff
@@ -1352,6 +1366,27 @@ func transformForRuntime(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec,
 	return nil
 }
 
+func transformDevicePluginForCDI(obj *appsv1.DaemonSet, config *gpuv1.ClusterPolicySpec, n ClusterPolicyController) {
+	if !config.CDI.IsEnabled() {
+		return
+	}
+
+	setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), CDIEnabledEnvName, "true")
+	setContainerEnv(&(obj.Spec.Template.Spec.Containers[0]), DeviceListStrategyEnvName, "envvar,cdi-annotations")
```

Comment:
Question: Under which conditions could we drop `envvar` and / or include `cdi-cri` here?

Reply:
Good question. 

I think we should drop envvar if native CDI is supported. 

As for `cdi-cri`, I think it depends on what versions of k8s we will support moving forward. The `CDIDevices` field in the CRI was alpha in 1.28 and beta since 1.29. So one could argue that we switch to `cdi-cri` altogether in this PR and never configure `cdi-annotations` as the device-list-strategy.

Reply:
What would the behavior be if we configure both `cdi-annotations` and `cdi-cri`? What is the behavior if we attempt to set the CRI field but we are running on k8s < 1.28?